client/lib/css-hot-reload/index.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+var io = require( 'socket.io-client' );
+
+/**
+ * Internal dependencies
+ */
+import { cssBuildFailed, cssBuilding } from 'state/application/actions';
+import i18n from 'lib/mixins/i18n';
+
+/**
+ * @returns {boolean} 
+ */
+function isChanged( href, changedFiles ) {
+	// "/calypso/style-debug.css?v=5a1db7fee7" -> "/calypso/style-debug.css"
+	var css = href.split('?')[0];
+	var i, re;
+		
+	for( i = 0; i < changedFiles.length; ++i ) {
+		re = new RegExp( '/' + changedFiles[i] + '$' );
+		if ( css.match( re ) ) {
+			return true;
+		}
+	}
+	
+	return false;
+}
+
+var CssHotReload = {
+
+	/**
+	 * Initialize client CSS hot-reload handler
+	 */
+	init: function( reduxStore ) {
+
+		var namespace = location.protocol + '//' + location.host + '/css-hot-reload';
+		var socket = io.connect( namespace );
+
+		socket.on( 'css-hot-reload', function( data ) {
+			switch( data.status ) {
+				case 'reload':
+					// Turn HTMLCollection to standard list
+					var elems = document.head.getElementsByTagName('LINK');
+					elems = [].slice.call( elems );
+					elems.forEach( function( oldLink ) { 
+						if ( ( 'href' in oldLink ) && isChanged( oldLink.href, data.changedFiles ) ) {
+							console.log( 'Reloading CSS: ', oldLink );
+							// Remove old .css and insert new one in the same spot
+							var newLink = document.createElement( 'LINK' );
+							oldLink.parentNode.insertBefore( newLink, oldLink );
+							oldLink.parentNode.removeChild( oldLink );
+							// Copy standard attributes
+							// https://developer.mozilla.org/en/docs/Web/HTML/Element/link
+							var attrs = [ 'crossorigin', 'href', 'hreflang',
```

Comment:
Two notes here: we could skip the variable declaration entirely; if we use `var` declarations, our style guide requires them at the top of the function because they are hoisted in fact. On the other hand, why not use `const` and `let` which are block-scoped?

```js
[ 
	'crossorigin', 
	'href', 
	'hreflang', 
	'media', 
	'rel',
	'sizes', 
	'title', 
	'type' 
].forEach( attr => {
	…
} );
```

I'll through this one in too just for a bit of a different way of seeing this problem…feel no obligation to make a change, but hopefully you will find some value in it.

```js
// defined way above in the module area
const standardAttributes = [ 
	'crossorigin', 
	'href', 
	'hreflang', 
	'media', 
	'rel',
	'sizes', 
	'title', 
	'type' 
];

const bustHashForHrefs = ( { name, oldValue } ) => {
	const value = 'href' === name
		? `${ oldValue.split( '?' ).shift() }?v=${ new Date().getTime() }`
		: value;

	return { name, value };
};

// inside init()
standardAttributes
	.filter( attr => oldLink.hasAttribute( attr ) )
	.map( attr => ( { name: attr, value: oldLink.getAttribute( attr ) } ) )
	.map( bustHashForHrefs );
	.forEach( ( { name, value } ) => newLink.setAttribute( name, value ) );
```